### PR TITLE
Final rename models changes

### PR DESCRIPTION
**Pull request by sima-zhu**
https://github.com/azure/azure-sdk-for-java/issues/11343

1. A leftover for DataSourceConnection
2. Builder suffix on SearchableField, SimpleField, ComplexField

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexerAsyncClient.java [hunk 1]
```diff
@@ -236,7 +236,7 @@ public String getEndpoint() {
      * help with debugging.
      * @return a list of DataSources
      */
-    public PagedFlux<SearchIndexerDataSourceConnection> listDataSourceConnection(RequestOptions requestOptions) {
+    public PagedFlux<SearchIndexerDataSourceConnection> listDataSourceConnections(RequestOptions requestOptions) {
```
Comment (sima-zhu) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexerAsyncClient.java:239]: @heaths Address the comments here.

File: sdk/search/azure-search-documents/CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,11 @@ and uplifted the properties to `SearchIndexerDataSourceConnection` and `SearchRe
 `NGramTokenFilter`, and `PathHierarchyTokenizer`.
 - Renamed `Similarity` to `SimilarityAlgorithm`.
 - Renamed `Suggester` to `SearchSuggester`.
+- Renamed fields `synonymMaps` to `synonymMapNames`, `analyzer` to `analyzerName`, 
+`searchAnalyzer` to `searchAnalyzerName` and `indexAnalyzer` to `indexAnalyzerName` 
+in `SearchField`, `SearchableField`.
+- Renamed `SimpleField` to `SimpleFieldBuilder`, `SearchableField` to `SearchableFieldBuilder`
+and `ComplexField` to `ComplexFieldBuilder`.
```
Comment (heaths) [inline sdk/search/azure-search-documents/CHANGELOG.md:25]: If you were specific on the `listDataSourceConnection` in the previous CHANGELOG, you should probably fix that one up as well.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SearchField.java [hunk 3]
```diff
@@ -688,11 +688,11 @@ public SearchField setIndexAnalyzer(LexicalAnalyzerName indexAnalyzer) {
      * attribute can be changed on existing fields. Must be null or an empty
      * collection for complex fields.
      *
-     * @param synonymMaps the synonymMaps value to set.
+     * @param synonymMapNames the synonymMaps value to set.
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SearchField.java:691]: Update javadoc to match the param name.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SearchableFieldBuilder.java [hunk 4]
```diff
@@ -31,19 +31,19 @@ public SearchableField(String name, boolean collection) {
...
+        return analyzerName;
     }
 
     /**
      * Sets the name of the language analyzer. This property cannot be set when either {@code searchAnalyzer} or
      * {@code indexAnalyzer} are set. Once the analyzer is chosen, it cannot be changed for the field in the index.
      *
-     * @param analyzer The {@link LexicalAnalyzerName} used for analyzer.
+     * @param analyzerName The {@link LexicalAnalyzerName} used for analyzer.
      * @return The SearchableField object itself.
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SearchableFieldBuilder.java:43]: Update javadoc return type in all methods of this class.
```suggestion
     * @return The SearchableFieldBuilder object itself.
```

**Pull request merged**